assembly-main/src/assembly/bin/che.sh

```diff
@@ -488,33 +488,56 @@ function stop_che_server {
     echo -e "Stopping Che server running on localhost:${CHE_PORT}"
     call_catalina >/dev/null 2>&1
   else
-    echo -e "Stopping Che server running in docker container named ${GREEN}che${NC}."
+    echo -e "Stopping Che server running in docker container named ${GREEN}che${NC}..."
 
     DOCKER_EXEC="sudo service docker stop && /home/user/che/bin/che.sh stop"
     "${DOCKER}" exec che ${DOCKER_EXEC} &>/dev/null 2>&1 || DOCKER_EXIT=$? || true
 
-    echo -e "Stopping docker container named ${GREEN}che${NC}."
+    echo -e "Stopping docker container named ${GREEN}che${NC}...."
     "${DOCKER}" stop che &>/dev/null 2>&1 || DOCKER_EXIT=$? || true
   fi
 }
 
 function kill_and_launch_docker_che {
 
-  echo -e "A Docker container for ${GREEN}che${NC} does not exist or duplicate conflict was discovered."
-  echo -e "Launching a new Docker container named ${GREEN}che${NC} from image ${GREEN}codenvy/che:${CHE_DOCKER_TAG}${NC} with Docker command:"
-  "${DOCKER}" kill che &> /dev/null || true
-  "${DOCKER}" rm che &> /dev/null || true
+  echo -e "A Docker container named ${GREEN}che${NC} does not exist or duplicate conflict was discovered."
+  echo -e "Cleaning up any zombie containers named ${GREEN}che${NC}..."
+
+  # Force remove any lingering zombie containers with the name che
+  "${DOCKER}" rm -f che &> /dev/null || true
+
+  echo -e "Pulling the Docker image labeled ${GREEN}codenvy/che:${CHE_DOCKER_TAG}${NC}..."
+  "${DOCKER}" pull codenvy/che:${CHE_DOCKER_TAG}
```

Comment:
If image exists locally but network is gone we will have failure her. Maybe add || true ?